### PR TITLE
fix: handle null return from iterator

### PR DESCRIPTION
**Pull request by bjagg**
##### Checklist

-   [x] the [individual contributor license agreement][] is signed
-   [x] commit message follows [commit guidelines][]
-   [ ] tests are included

##### Description of change

[individual contributor license agreement]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#individual-contributor-license-agreement
[commit guidelines]: https://github.com/Jasig/uPortal/blob/master/CONTRIBUTING.md#commit
[message properties]: https://github.com/Jasig/uPortal/tree/master/uportal-war/src/main/resources/properties/i18n
[WCAG 2.0 AA]: https://www.w3.org/WAI/WCAG20/quickref/?levels=aaa&technologies=smil%2Cpdf%2Cflash%2Csl

### CODE REVIEW

File: uPortal-groups/uPortal-groups-core/src/main/java/org/apereo/portal/groups/ReferenceIndividualGroupService.java [hunk 1]
```diff
@@ -175,17 +175,22 @@ public Iterator findParentGroups(IGroupMember gm) throws GroupsException {
         log.debug("Finding containing groups for member {}", gm.getKey());
         Collection groups = new ArrayList(10);
         IEntityGroup group = null;
-        for (Iterator it = getGroupStore().findParentGroups(gm); it.hasNext(); ) {
+        Iterator it = getGroupStore().findParentGroups(gm);
+        while (it.hasNext()) {
```
Comment (ChristianMurphy) [inline uPortal-groups/uPortal-groups-core/src/main/java/org/apereo/portal/groups/ReferenceIndividualGroupService.java:179]: Out of general curiosity, why was `for (item : iterable)` used elsewhere, but `while` used here?

**Review by ChristianMurphy — APPROVED** [review]
Thanks @bjagg!

File: uPortal-groups/uPortal-groups-core/src/main/java/org/apereo/portal/groups/ReferenceIndividualGroupService.java (hunk 1 above)
Comment (bjagg) [reply uPortal-groups/uPortal-groups-core/src/main/java/org/apereo/portal/groups/ReferenceIndividualGroupService.java:179]: It is to signify when the collection is the right class verses when the collection is being cast. If I had more time, I would consider making it consistent across the class.

Comment (ChristianMurphy) [reply uPortal-groups/uPortal-groups-core/src/main/java/org/apereo/portal/groups/ReferenceIndividualGroupService.java:179]: thanks for the clarification, totally get the need to timebox.

**Pull request merged**